cmake/Modules/CocosConfigDefine.cmake

```diff
@@ -0,0 +1,185 @@
+     else()
+         set(COMPILER_STRING "GCC")
+     endif()
+ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+     set(COMPILER_STRING "${CMAKE_CXX_COMPILER_ID} C++")
+ elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+     set(COMPILER_STRING "Visual Studio C++")
+ else()
+     set(COMPILER_STRING "${CMAKE_CXX_COMPILER_ID}")
+ endif()
+ if(CMAKE_CROSSCOMPILING)
+     set(BUILDING_STRING "It appears you are cross compiling for ${SYSTEM_STRING} with ${COMPILER_STRING}")
+ else()
+     set(BUILDING_STRING "It appears you are building natively for ${SYSTEM_STRING} with ${COMPILER_STRING}")
+ endif()
+ message(STATUS ${BUILDING_STRING})
+
+# check c++ standard
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# check visual studio version
+ if(WINDOWS)
+    # not support other compile tools except MSVC for now
+    if(MSVC)
+        # Visual Studio 2015, MSVC_VERSION 1900      (v140 toolset)
+        # Visual Studio 2017, MSVC_VERSION 1910-1919 (v141 toolset)
+        if(${MSVC_VERSION} EQUAL 1900 OR ${MSVC_VERSION} GREATER 1900)
+            message(STATUS "using Windows MSVC generate cocos2d-x project, MSVC_VERSION:${MSVC_VERSION}")
+        else()
+            message(FATAL_ERROR "using Windows MSVC generate cocos2d-x project, MSVC_VERSION:${MSVC_VERSION} lower than needed")
+        endif()
+    else()
+        message(FATAL_ERROR "please using Windows MSVC compile cocos2d-x project, support other compile tools not yet")
+    endif()
+endif()
+
+ # Set macro definitions for special platforms
+ function(use_cocos2dx_compile_define target)
+    target_compile_definitions(${target} PUBLIC "COCOS2D_DEBUG$<$<CONFIG:Debug>:=1>")
+    if(APPLE)
+        target_compile_definitions(${target} PUBLIC __APPLE__)
+        target_compile_definitions(${target} PUBLIC USE_FILE32API)
+    elseif(LINUX)
+        target_compile_definitions(${target} PUBLIC LINUX)
+        target_compile_definitions(${target} PUBLIC _GNU_SOURCE)
+    elseif(ANDROID)
+        target_compile_definitions(${target} PUBLIC ANDROID)
+        target_compile_definitions(${target} PUBLIC USE_FILE32API)
+    elseif(WINDOWS)
+        target_compile_definitions(${target} 
+            PUBLIC WIN32
+            PUBLIC _WIN32
+            PUBLIC _WINDOWS
+            PUBLIC UNICODE
+            PUBLIC _UNICODE
+            PUBLIC _CRT_SECURE_NO_WARNINGS
+            PUBLIC _SCL_SECURE_NO_WARNINGS
+        )
+        if(BUILD_SHARED_LIBS)
+            target_compile_definitions(${target} 
+                PUBLIC _USRDLL
+                PUBLIC _EXPORT_DLL_
+                PUBLIC _USEGUIDLL
+                PUBLIC _USREXDLL
+                PUBLIC _USRSTUDIODLL
+                PUBLIC _USE3DDLL
+            )
+        else()
+            target_compile_definitions(${target} PUBLIC CC_STATIC)
+        endif()
+    endif()
+endfunction()
+
+ # Set compiler options
+ function(use_cocos2dx_compile_options target)
```

Comment:
I think that this function should be removed. Proper default values are set by cmake (e.g. -g for Debug, RelWithDebInfo). 
With those settings user can not change it.

Reply:
After the CI passed, I could try to disable this function to see whether default values works

Reply:
compile options have been simplified, I don't think we can remove this function, the default values possible works, but may not proper for cocos2d-x project. 

when I remove this function, and then generate VS project to test, I found `/MP` isn't been set, so that the compile process will works in single thread, spend a long time to compile.